### PR TITLE
Exchange: save form data for last session

### PR DESCRIPTION
**Pull request by NotEternal**
## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/swaponline/MultiCurrencyWallet/blob/master/docs/CONTRIBUTING.md) guide
- [x] Good naming, keep simple
- [x] Tested desktop/mobile
- [x] Tested bright/dark
- [x] Tested en/ru
- [ ] Affects money; I checked the functionality once again
- [x] I checked the PR once again

### Original issue
#3989

### CODE REVIEW

File: src/front/shared/pages/Exchange/Exchange.tsx [hunk 1]
```diff
@@ -191,33 +182,81 @@ const bannedPeers = {} // rejected swap peers
...
+   * @method handleSetHaveValue
+   * @method applyAddress
+   * @method flipCurrency
+   * @method resetState
+   * @method chooseCurrencyToRender
+   * @method extendedControlsSet
+   * @method doesComissionPreventThisOrder
+   * @method goDeclimeFaq
+   * @method showIncompleteSwap
+   */
```
Comment (iower) [inline src/front/shared/pages/Exchange/Exchange.tsx:241]:
```suggestion
```

Generated? JSDoc seems not very helpful.

**Review by iower — COMMENTED** [review]
Suggestions:
- `exchangeData` -> `exchangeSettings`

**Review by iower — CHANGES_REQUESTED** [review]
before / after (worse)
image

- [x]  fixed

newfix

File: src/front/shared/pages/Exchange/Exchange.tsx (hunk 1 above)
Comment (NotEternal) [reply src/front/shared/pages/Exchange/Exchange.tsx:241]: For large files, it's better when all methods in one place

Comment (NotEternal) [reply src/front/shared/pages/Exchange/Exchange.tsx:241]: agree, will be better make docs for more important code

**Comment by iower** [comment]
Suggestion:
- `exchangeData` -> `exchangeSettings` (because `exchangeData` associated with offers/prices)

File: package.json [hunk 2]
```diff
@@ -120,6 +111,7 @@
     "babel-plugin-module-resolver": "^4.1.0",
     "babel-plugin-react-intl": "^8.2.25",
     "basic-auth": "^2.0.1",
+    "bestzip": "^2.1.7",
```
Comment (iower) [inline package.json:114]:
```suggestion
```

**Review by iower — APPROVED** [review]

**Pull request merged**